### ISSUE TITLE
[BUG] Mails with neural scores stuck in greylist.

### ISSUE THREAD

**Issue by whataboutpereira** [rspamd/rspamd#3232]
### Prerequisites

* [x] Put an X between the brackets on this line if you have done all of the following:
    * Read about bug reporting in general: https://rspamd.com/doc/faq.html#how-to-report-bugs-found-in-rspamd
    * Enabled relevant debugging logs: https://rspamd.com/doc/faq.html#how-to-debug-some-module-in-rspamd 
    * Checked the FAQs about Core files in case of fatal crash: https://rspamd.com/doc/faq.html#how-to-figure-out-why-rspamd-process-crashed
    * Tried ASAN package and obtained the ASAN report (if possible): https://rspamd.com/doc/faq.html#asan-builds 
    * Checked that your issue isn't already filed: https://github.com/issues?utf8=%E2%9C%93&q=is%3Aissue+user%3Arspamd
    * Checked that there is not already an experimental package or master branch

**Describe the bug**
I'm seeing mails that are permanently greylisted. It started when neural scores started being applied. All these mails have a score below by greylisting limit and go over the limit after neural score is added.

### Steps to Reproduce

1. Receive a mail with a score just below greylist limit
2. Apply a positive neural score that pushes the score over greylist limit

**Expected behavior**
Greylisted messages with neural scores should be let through after their greylisting period passes.

### Versions

Rspamd v2.1
CentOS 7

### Additional Information

Example mail with greylist limit set at 6.
Score: 6.21 / 15

```
MIME_MA_MISSING_TEXT (2)
HTML_SHORT_LINK_IMG_1 (2)
NEURAL_SPAM_SHORT (1.377817) [0.689]
URI_COUNT_ODD (1) [21]
FORGED_SENDER (0.3) [notifications@updates.bndsintown.com,bounces@sg.updates.bndsintown.com]
RBL_SENDERSCORE_EXCELLENT (-0.2) [167.89.19.190:from]
R_SPF_ALLOW (-0.2) [+ip4:167.89.19.190]
R_DKIM_ALLOW (-0.2) [updates.bndsintown.com:s=s1]
MIME_HTML_ONLY (0.2)
MIME_GOOD (-0.1) [multipart/alternative]
ONCE_RECEIVED (0.1)
BAYES_HAM (-0.071237) [60.44%]
FROM_NEQ_ENVFROM (0) [notifications@updates.bndsintown.com,bounces@sg.updates.bndsintown.com]
ARC_SIGNED (0) [i=1]
DKIM_TRACE (0) [updates.bndsintown.com:+]
ASN (0) [asn:11377, ipnet:167.89.0.0/18, country:US]
RCVD_IN_DNSWL_NONE (0) [167.89.19.190:from]
RCVD_TLS_ALL (0)
MIME_TRACE (0) [0:+,1:~]
RCVD_COUNT_ONE (0) [1]
PREVIOUSLY_DELIVERED (0) [r@m.ee]
ARC_NA (0)
DMARC_NA (0) [bndsintown.com: no valid DMARC record]
TO_DN_NONE (0)
TAGGED_FROM (0) [7507462-eb99-r=m.ee]
FROM_HAS_DN (0)
RCPT_COUNT_ONE (0) [1]
RWL_MAILSPIKE_POSSIBLE (0) [167.89.19.190:from]
TO_MATCH_ENVRCPT_ALL (0)
```

**Comment by vstakhov** [comment]
There is a significant difference between actions and their thresholds. You are talking about thresholds as far as I see. After greylisting period passed greylisting threshold can not be applied and it does not depend on anything else (including neural network).

**Comment by whataboutpereira** [comment]
The greylist timeout is set at the default 5min, expire at the default 1d. The mails are being resent every hour and they are greylisted constantly. What else can I check?

**Comment by whataboutpereira** [comment]
I now also notice that the logs first state that the score is too low for greylisting and then they become greylisted anyway. Without the slightly positive neural score they wouldn't get greylisted.

```
2020-01-23 10:08:59 #1887(normal) <b139fe>; lua; greylist.lua:298: Score too low - skip greylisting
2020-01-23 10:08:59 #1887(normal) <b139fe>; task; rspamd_task_write_log: id: <sry5YnKHSmC4C_hHiecesw@ismtpd0013p1iad2.sendgrid.net>, qid: <46be4d92>, ip: 167.89.19.190, from: <bounces@sg.updates.bndsintown.com>, (default: T (greylist): [6.27/15.00]
```

**Comment by whataboutpereira** [comment]
Another difference I notice is that they problem mails get:
default: T (greylist)

Greylisted mails with a slightly higher score get:
default: F (soft reject)

What's the significance between the T and F?

I don't see an action "greylist" defined anywhere in my Rspamd config files.

**Comment by whataboutpereira** [comment]
I restarted Rspamd which wiped the neural data and the mail now came throught with a score of 5.90 / 15.

**Comment by vstakhov** [comment]
Again, you mix scores and thresholds. `greylist` should never ever be returned in a reply: you clearly do something wrong. `soft reject` makes sense. `T` and `F` are just legacy from SA compatibility, ignore that crap.

**Comment by whataboutpereira** [comment]
I agree default: T (greylist) doesn't make sense especially as it has just logged that greylisting is skipped.

How can I completely disable greylisting?

I these months I've used Rspamd I've yet to see a spam mail not be resent if it's greylisted. I tried setting enabled = false in local.d/greylist.conf, but that actually made it greylist everything above threshold.

The only place in my local.d greylist is mentioned:

/etc/rspamd/local.d/actions.conf:
```
greylist   = 6;
```
/etc/rspamd/local.d/settings.conf for a rcpt that receives spam:
```
actions {
    reject = 100.0;
    greylist = null;
    "add header" = 0.0;
}
```
